### PR TITLE
Internal steps && advanced quickstarts

### PR DESCRIPTION
**Pull request by Tonnu**
Contains:
* Internal step docs
* Advanced quickstarts section
* Article on building minimal containers with Go

### CODE REVIEW

**Comment by lindseybateman** [comment]
nice

File: content/docs/steps/internal-steps.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
+## Internal steps
+Internal steps are developed by wercker and are baked into the wercker cli as
+these interact with the Docker API that is external from the container. From a
+security perspective we don’t want to make this funcionality available from
+inside the Docker container, and as such have created these internal steps.
```
Comment (hatchan) [inline content/docs/steps/internal-steps.md]: This is not from a security perspective, but from a technical perspective. It is not possible to control the docker daemon from inside the container.

File: content/docs/steps/internal-steps.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+deploy:
+  steps:
+    - internal/docker-push:
+        username: $USERNAME
+        password: $PASSWORD
+        tag: my-amazing-tag
+        repository: turing/bar
+        registry: https://registry.hub.docker.com
+```
+
```
Comment (hatchan) [inline content/docs/steps/internal-steps.md]: Maybe add a short, single line/paragraph what this step does.

File: content/docs/steps/internal-steps.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+More information about the internal/docker-push step can be found
+[here](/docs/containers/pushing-containers.html).
+
+### <a name="scratch-push" class="anchor"></a>internal/docker-scratch-push
+The `docker-scratch-push` step works the same as the normal `docker-push` step.
+The main difference is that this step uses a
+[scratch](https://docs.docker.com/articles/baseimages/) base image provided by
+Docker. This image contains only the bare essentials, and as such is very
+lightweight. To set it up correctly however, it requires some additional steps. 
+
```
Comment (hatchan) [inline content/docs/steps/internal-steps.md]: I would like to see what gets injected into the container, and where. (ie. either the source dir or output dir).

File: content/docs/steps/internal-steps.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        code: node app.js
+        reload: true
+```
+
+With this dev pipeline we run one initial setup step, `npm-install`, to prepare
+our container's environment, then we execute our node app, reloading on changes.
+
+To run this, you would do:
+
+ `$ wercker dev --publish 5000`
```
Comment (hatchan) [inline content/docs/steps/internal-steps.md]: use the tripple backticks + `no-highlight`. And remove the `$`.

Comment (hatchan) [inline content/docs/steps/internal-steps.md]: See #250

File: content/docs/steps/internal-steps.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+And once it loads, browse to your docker host (either localhost on linux, or
+with boot2docker usually  on 192.168.59.103) on port 5000 to see your app running.
+For your convenience, we'll tell you the IP once the step runs.
+
+As you make changes to your code, the app will be reloaded, but the npm-install
+steps will not be run again.
+
+Without the "reload: true" the code will only be run once, which is useful if
+your development server has its own reloading semantics (or is only loading
+static files).
```
Comment (hatchan) [inline content/docs/steps/internal-steps.md]: Worth mentioning that the step (and the next step) only apply to local builds

Comment (hatchan) [inline content/docs/steps/internal-steps.md]: Also might be a good idea to link to the dev article (and vice versa)

File: content/quickstarts/advanced/building-minimal-containers-with-go.md [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+statically compile our application. We disable the ability to create Go
+packages that call C code with the `CGO_ENABLED=0` flag, rebuild all
+dependencies with the `-a` flag, and finally we remove any debug information.
+
+The last step copies the executable into the
+`$WERCKER_OUTPUT_DIR` folder, and the contents of this folder gets passed along
+to the `/pipeline/source/` folder within the deploy pipeline.
+
+We're now ready to push our container to a registry - in this case
+[quay.io](https://quay.io).
```
Comment (hatchan) [inline content/quickstarts/advanced/building-minimal-containers-with-go.md]: Capitalize to `Quay.io`

File: content/quickstarts/advanced/building-minimal-containers-with-go.md [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+
+### Deploy Pipeline
+
+The first step is to create a new deploy target for our application on
+https://app.wercker.com. This deploy target will hold the necessary information
+that is needed to deploy to quay.io.
```
Comment (hatchan) [inline content/quickstarts/advanced/building-minimal-containers-with-go.md]: Capitalize to `Quay.io`

File: content/quickstarts/advanced/building-minimal-containers-with-go.md [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+        password: $QUAY_PASSWORD
+        cmd: ./app
+        tag: $WERCKER_GIT_COMMIT
+        ports: "5000"
+        repository: quay.io/<your-repo>
+        registry: https://quay.io
+```
+
+The deploy section of our
+[wercker.yml](http://devcenter.wercker.com/docs/wercker-yml/index.html) above
```
Comment (hatchan) [inline content/quickstarts/advanced/building-minimal-containers-with-go.md]: leave out the host part

**Comment by hatchan** [comment]
minor comments, other than that :+1:

**Pull request merged**